src/operations/transforms/mirror.js

```diff
@@ -0,0 +1,53 @@
+const flatten = require('../utils/flatten')
+
+const mat4 = require('../../math/mat4')
+const plane = require('../../math/plane')
+
+const {geom2, geom3, path2} = require('../../geometry')
+
+/**
+ * Mirror the given object(s) using the given options (if any)
+ * Note: The normal should be given as 90 degrees from the plane origin.
+ * @param {Object} options - options for mirror
+ * @param {Array} options.origin=[0,0,0] - the origin of the plane
+ * @param {Array} options.normal=[0,0,1] - the normal vector of the plane
+ * @param {Object|Array} objects - the objects(s) to mirror
+ * @return {Object|Array} the mirrored object(s)
+ *
+ * @example
+ * let newsphere = mirror({normal: [0,0,10]}, cube({center: [0,0,15], radius: [20, 25, 5]}))
+ */
+const mirror = (options, ...objects) => {
+  const defaults = {
+    origin: [0, 0, 0],
+    normal: [0, 0, 1] // Z axis
+  }
+  let {origin, normal} = Object.assign({}, defaults, options)
+
+  objects = flatten(objects)
+  if (objects.length === 0) throw new Error('wrong number of arguments')
+
+  const planeOfMirror = plane.fromNormalAndPoint(normal, origin)
+  const matrix = mat4.mirrorByPlane(planeOfMirror)
+
+  const results = objects.map((object) => {
+    if (path2.isA(object)) return path2.transform(matrix, object)
+    if (geom2.isA(object)) return geom2.transform(matrix, object)
+    if (geom3.isA(object)) return geom3.transform(matrix, object)
+    return object
+  })
+  return results.length === 1 ? results[0] : results
+}
+
+const mirrorX = (...objects) => mirror({normal: [1, 0, 0]}, objects)
```

Comment:
this connects back to our previous questioning: do we need mirrorX, Y , Z ? I think we could get by without them right ? (there is barely any gain , verbosity wise), perhaps we can keep them for now but flag them as candidates for removal for V2 final, no ?

Reply:
Let's keep the mirrorX/Y/Z for now, as V1 has those functions. I have used them in designs as well.

Reply:
but is one of the purposes of V2 not to get rid of the clutter ? I also used it in some of my designs, but replacing it with a normal is trivial ?